shoppingfeed.php

```diff
@@ -629,31 +629,37 @@ public function mapPrestashopProduct($sfProductReference, $id_shop, ...$argument
      */
     public function mapProductPrice(ShoppingfeedProduct $sfProduct, $id_shop, $arguments = [])
     {
-        $cloneContext = Context::getContext()->cloneContext();
-        $cloneContext->shop = new Shop($id_shop);
-
         $specific_price_output = null;
         Product::flushPriceCache();
-        $price = Product::getPriceStatic(
-            $sfProduct->id_product, // id_product
-            true, // usetax
+
+        // Tax depends on a country. We use country configured as default one for shop.
+        $id_country = Configuration::get('PS_COUNTRY_DEFAULT', null, null, $id_shop);
+        $id_currency = Validate::isLoadedObject($this->context->currency) ? (int) $this->context->currency->id : (int) Configuration::get('PS_CURRENCY_DEFAULT', null, null, $id_shop);
+        $id_group = (int) Group::getCurrent()->id;
+
+        $price = Product::priceCalculation(
+            $id_shop,
+            $sfProduct->id_product,
             $sfProduct->id_product_attribute ? // id_product_attribute
                 $sfProduct->id_product_attribute : null,
-            2, // decimals
-            null, // divisor
-            false, // only_reduc
+            $id_country,
+            0,
```

Comment:
Could you please comment values with parameter name like previous method ?
It's better to read and debug, this method has too many parameters.